packages/devtools_app/lib/src/service_extensions.dart

```diff
@@ -51,7 +52,8 @@ class ToggleableServiceExtensionDescription<T>
 
 class ServiceExtensionDescription<T> {
   ServiceExtensionDescription({
-    this.icon,
+    this.disabledIcon,
```

Comment:
should these both be required fields?

Reply:
Service extensions `httpEnableTimelineLogging` and `socketProfiling` does not have any kind of icon.  Either we need to add an icon for those or keep icon not required. Do we have any icons that could be used?

Reply:
It seems that they are not under any button and are called in network recording. So could I just add some icons for them to be used in the future if needed, but for now just to please required for enabled and disabled icon?

For the icons, what about `Icons.timeline` for `httpEnableTimelineLogging` and `Icons.bar_chart` for `socketProfiling`? 
![Screenshot 2021-07-01 at 14 25 26](https://user-images.githubusercontent.com/17140689/124116940-367ca080-da78-11eb-9220-7d3b3ae79d53.png)
![Screenshot 2021-07-01 at 14 35 36](https://user-images.githubusercontent.com/17140689/124118225-c4a55680-da79-11eb-9f62-c3b69f43e457.png)

Reply:
Or one option is to just add Container-widget or Placeholder-widget for it, since the type for it is a Widget

Reply:
Let's just make both of these String consts with their extension values `'ext.dart.io.httpEnableTimelineLogging'` and `'ext.dart.io.socketProfilingEnabled'`. This is all we use them for right now anyway

Reply:
I think we can't do that, because `httpEnableTimelineLogging` and `socketProfilingEnabled` needs to be in `Map<String, ServiceExtensionDescription> serviceExtensionsAllowlist`. Which is used to do extension checks in multiple places even for `httpEnableTimelineLogging` and `socketProfilingEnabled`. For now I will put Placeholder() as `enabledIcon` to pass the required parameter, is it okay or am I missing something?

Reply:
We should add these extensions to serviceExtensionAllowlist probably adding extra functionality to the related logic for serviceExtensionAllowlist to handle the fact that these extensions should be triggered for all isolates not just the main isolate.